### DISCUSSION TITLE
Linter for limit param number?

### DISCUSSION THREAD

**Discussion by caibirdme** [golangci/golangci-lint#5068] [Q&A] [answered]
Are there any linters that can check if a func has too many params?
I want to forbid code like this:
```go
func hello(a,b,c,d,e,f,g,h,i,j,k,m,l,n int) {}
```
or something like:
```go
func hi() (int,int,int,int,int,int,int,int,error) {}
```

Am I missing some existed linter?

**Comment by webstradev** [comment, marked as answer]
This already exists within the revive linter.

**Reply by bombsimon** [reply]
To elaborate, the rule is called `argument-limit` ([docs](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#argument-limit)), example here:

https://github.com/golangci/golangci-lint/blob/b91b32470acb2e04486f80010356212f847ccfd0/.golangci.reference.yml#L1712-L1717
